resources/PythonPip3Dsc/PythonPip3Dsc.psm1

```diff
@@ -205,4 +248,181 @@ function Invoke-Pip3
     }
 }
 
-#endregion Functions
\ No newline at end of file
+function TryGetRegistryValue{
+    param (
+        [Parameter(Mandatory = $true)]
+        [string]$Key,
+
+        [Parameter(Mandatory = $true)]
+        [string]$Property
+        )    
+
+    if (Test-Path -Path $Key)
+    {
+        try
+        {
+            return (Get-ItemProperty -Path $Key | Select-Object -ExpandProperty $Property)     
+        }
+        catch
+        {
+            Write-Verbose "Property `"$($Property)`" could not be found."
+        }
+    }
+    else
+    {
+        Write-Verbose "Registry key does not exist."
+    }
+}
+
+#endregion Functions
+
+$global:usePip3Exe = $false
+$global:pip3ExePath = Get-Pip3Path
+
+# Assert once that pip3 is already installed on the system.
+Assert-Pip3
+
+#region DSCResources
+<#
+.SYNOPSIS
+    The `Pip3Package` DSC Resource allows you to install, update, and uninstall Python packages using pip3.
+
+.PARAMETER SID
+    The security identifier. This is a key property and should not be set manually.
+
+.PARAMETER Exist
+    Indicates whether the package should exist. Defaults to $true.
+
+.PARAMETER Package
+    The name of the Python package to manage. This is a mandatory property.
+
+    For a list of Python packages, see https://pypi.org/.
+
+.PARAMETER Version
+    The version of the Python package to manage. If not specified, the latest version will be used.
+
+.PARAMETER Arguments
+    Additional arguments to pass to pip3.
+
+.PARAMETER InstalledPackages
+    A list of installed packages. This property is not configurable.
+
+.EXAMPLE
+    PS C:\> Invoke-DscResource -ModuleName PythonPip3Dsc -Name Pip3Package -Method Set -Property @{ Package = 'flask' }
+
+    This example installs the Flask package.
+
+.EXAMPLE
+    PS C:\> Invoke-DscResource -ModuleName PythonPip3Dsc -Name Pip3Package -Method Get -Property @{ Package = 'flask'; Version = '1.1.4' }
+
+    This example shows how to get the current state of the Flask package with version. If the version is not found, the latest version will be used if flask is found.
+
+.EXAMPLE
+    PS C:\> Invoke-DscResource -ModuleName PythonPip3Dsc -Name Pip3Package -Method Get -Property @{ Package = 'django'; Exist = $false }
+
+    This example shows how Django can be removed from the system.
+#>
+[DSCResource()]
+class Pip3Package
+{
+    # DSCResource requires a key. Do not set.
+    [DscProperty(Key)]
+    [string]$SID
```

Comment:
This is a mistake I made but I think `Key` can be moved over to the $Package parameter since that name should be always included.